corehq/apps/hqadmin/couch_domain_utils.py

```diff
@@ -51,21 +59,34 @@ def cleanup_stale_es_on_couch_domains(
                 }
             )
             if stdout:
-                stdout.write(f"\nFound {len(forms_not_in_es)} forms not in es "
+                stdout.write(f"\tFound {len(forms_not_in_es)} forms not in es "
                              f"for {domain}")
             changes = _get_changes(domain, forms_not_in_es)
             form_es_processor = get_xform_pillow().processors[0]
             for change in changes:
                 form_es_processor.process_change(change)
 
 
-def _get_couch_form_ids(domain, start, end):
-    "Get form IDs from the 'by_domain_doc_type_date/view' couch view"
+def _get_couch_node_databases(db, node_port=COUCH_NODE_PORT):
+    def node_url(proxy_url, node):
+        return urlunparse(proxy_url._replace(netloc=f'{auth}@{node}:{node_port}'))
+
+    resp = db.server._request_session.get(urljoin(db.server.uri, '/_membership'))
+    resp.raise_for_status()
+    membership = resp.json()
+    nodes = [node.split("@")[1] for node in membership["cluster_nodes"]]
+    proxy_url = urlparse(settings.COUCH_DATABASE)._replace(path=f"/{db.dbname}")
+    auth = proxy_url.netloc.split('@')[0]
+    return [Database(node_url(proxy_url, node)) for node in nodes]
```

Comment:
@biyeun I'm sorry this knowledge didn't get passed along but we discovered that querying individual nodes has no advantage since CouchDB will internally distribute the query among the nodes.

Reply:
Sorry, somewhat of a duplicate comment to @millerdev